### PR TITLE
Add more friendly error messages (WIP)

### PR DESCRIPTION
**Pull request by echochamber**
So I learned a bit playing with this crate seeing if I could fix issue #49.

Here's what I got screen shot 2017-02-18 at 1 00 16 am

This PR right now just demonstrates a nicer way to surface the error in one place. I can apply this same way of surfacing the error across the rest of the crate where we currently have panics.

The reason it is still panicing is because `Parse::parse()` still requires a return value of self, and since parse_block failed to return an `syntax::ast::Block`, I don't have a value to use for `Test::block`.

@reem I was hoping to get your opinion on what your preferred way to apply this change would be. This is my first time working with syntax extensions and I don't have much experience writing syntax parsers in general.

Options I see (but there certainly could be some I'm overlooking):

* Make `Test::block` an option type, make it none when `parse_block()` fails. Although I'm not sure how Generate::generate should behave when block is `None`
* Still panic anytime a parse fails, but add this friendlier way of displaying the error beforehand. This is really ugly and hacky but easiest to implement and still gives the user the info they need to locate the issue.
* Make Parse::parse return an Option<Self>. Not really sure about the implications of this one. Again, parsing and syntax extensions is really not my area of expertise.

I think I can implement any of the 3 myself, but would definitely appreciate a bit of insight beforehand.

### CODE REVIEW

File: tests/my_test.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#![feature(plugin,const_fn)]
```
Comment (echochamber) [inline tests/my_test.rs]: This is just here for demonstration, I'll remove it before merging.

**Comment by echochamber** [comment]
@reem Not to be a pest, but have you had a chance to look at my comment above yet?

**Comment by ujh** [comment]
Sorry about the long wait! I'm the new maintainer and I will have a look at this soon.

**Comment by echochamber** [comment]
@ujh Thanks for the update.

**Comment by ujh** [comment]
Could you rebase on the current `master`? Then the tests would probably pass again and I can have a closer look at what you did.

**Comment by echochamber** [comment]
@ujh Rebased. Let me know if there's anything else I can do to assist :)

**Comment by ujh** [comment]
Sorry for the delay! I'm OK with the may it's implemented right now or with whatever you think is best. Let's make this user friendly and let's worry about maintenance later.

Right now you know more about this crate than I do. It's something I want to change soon(ish) but for now just do whatever you think is best.

**Comment by echochamber** [comment]
@ujh Sounds good. I'll revisit this again before deciding to refresh on my findings since I wrote this PR over a month ago.

If you are interested in what I learned to get a bit more up to speed I'll explain (apologies I'm explaining something you already know). Basically what I found was that during the macro expansion step, the custom macros in this crate make `panic!()` calls when encountering invalid syntax. In the case of the issue ticket this PR is for it was:

`it "fails to parse"()`

Whereas stainless doesn't expect any parenthesis there:

`it "fails to parse"`

What I learned was that when there is a panic during macro expansion (or in the compilation step in general), the compiler doesn't get a chance to show any information about the encountered error. Instead the correct way to handle an error in a custom macro is to emit the error like this `err.emit();`. This call to emit actually shows useful debugging info:

emmited


Whereas without the emit and just panicing displays this (which is what we currently do):

panic

**Pull request closed without merging**